stdlib/Distributed/test/distributed_exec.jl

```diff
@@ -143,9 +143,8 @@ function test_futures_dgc(id)
     @test fetch(f) == id
     @test f.v !== nothing
     yield(); # flush gc msgs
```

Comment:
I believe this yield is not correct, same with the yield on the other side.
Nothing guarantees that the flush gc_msgs task has run. 

Same with the yield on the other side.
@vtjnash any ideas on redeeming this test, short of yanking it out?

Reply:
Probably can't do much better than loop the test below for a couple seconds, and see if it finishes within that time

Reply:
@DilumAluthge I think what Jameson was saying was something like

```julia
@test begin
    t0 = time_ns()
    while remotecall_fetch(k->(yield();haskey(Distributed.PGRP.refs, k)), id, fid) != false
        sleep(0.1)
        if (time_ns() - t0) / 1e9 > 10  # wait 10s
            error("timeout")
        end
    end
    true
end
```

Reply:
Ahhh, I see. I've pushed a new commit that implements your code. I went ahead and did a 60 second timeout.